src/hotspot/share/code/scopeDesc.cpp

```diff
@@ -114,7 +114,6 @@ GrowableArray<ScopeValue*>* ScopeDesc::decode_object_values(int decode_offset) {
     // object's fields could reference it (OBJECT_ID_CODE).
     (void)ScopeValue::read_from(stream);
   }
-  assert(result->length() == length, "inconsistent debug information");
```

Comment:
Is this assert no longer correct? Is it expected now that `result->length()` might not equal `length` in some cases?

Reply:
I had commented this assert and added a comment in it's place, but I removed it during the previous review round.

The comparison is no longer valid because result will also hold "candidate" objects and "length" is only the number of non-candidate objects.